RecoJets/JetProducers/src/MVAJetPuId.cc

```diff
@@ -0,0 +1,561 @@
+#include "RecoJets/JetProducers/interface/MVAJetPuId.h"
+#include "DataFormats/JetReco/interface/PFJet.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "DataFormats/JetReco/interface/Jet.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "CommonTools/Utils/interface/TMVAZipReader.h"
+#include "TMatrixDSym.h"
+#include "TMatrixDSymEigen.h"
+#include "DataFormats/JetReco/interface/PileupJetIdentifier.h"
+
+const float large_val = std::numeric_limits<float>::max();
+
+
+MVAJetPuId::MVAJetPuId(const edm::ParameterSet & ps) 
+{
+	impactParTkThreshod_ = 1.;
+
+	tmvaWeights_         = edm::FileInPath(ps.getParameter<std::string>("tmvaWeights")).fullPath(); 
+	tmvaMethod_          = ps.getParameter<std::string>("tmvaMethod");
+	tmvaVariables_       = ps.getParameter<std::vector<std::string> >("tmvaVariables");
+	tmvaSpectators_      = ps.getParameter<std::vector<std::string> >("tmvaSpectators");
+	version_             = ps.getParameter<int>("version");
+	reader_              = 0;
+	edm::ParameterSet jetConfig = ps.getParameter<edm::ParameterSet>("JetIdParams");
+	for(int i0 = 0; i0 < 3; i0++) { 
```

Comment:
the constants NWPs etc were defined to be used in the loops as well, not just to define the array size in the header file.
Please use them here